backend/internal/controller/server/msg_handler.go

```diff
@@ -66,3 +66,60 @@ func (s server) createResponse(reader io.Reader) ([]byte, error) {
 func (s server) ResetScoreHandler(w http.ResponseWriter, r *http.Request) {
 	s.game.ResetScore()
 }
+
+func (s server) SendScoreHandler(w http.ResponseWriter, r *http.Request) {
+
+	closeConnChan := make(chan bool)
+
+	var upgrader websocket.Upgrader
+	// TODO: We should check the origin in the future. For now we enable every connection to the server.
+	upgrader.CheckOrigin = func(r *http.Request) bool { return true }
+	c, err := upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		log.Println(err)
+	}
+
+	defer c.Close()
+
+	if err := s.writeScoreMessage(c); err != nil {
+		log.Println(err)
+	}
+
+	go waitForError(c, closeConnChan)
+
+	for {
+		select {
+		case <-s.game.GetChannel():
+			if err := s.writeScoreMessage(c); err != nil {
+				log.Println(err)
+				break
+			}
+		case <-closeConnChan:
+			return
+
+		}
+	}
+}
+
+func (s *server) writeScoreMessage(c *websocket.Conn) error {
+	gameScoreMsg, err := json.Marshal(s.game.GetScore())
```

Comment:
This can cause data races. Game should be protected with a mutex + I'd send game score over the event channel to avoid mutex locking

Reply:
Done